### PR TITLE
px 1.4.0 (new formula)

### PR DESCRIPTION
**Pull request by ritzmann**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

The formula uses the git strategy to download the sources because setup.py from upstream will fail if it cannot execute `git describe`.

### CODE REVIEW

File: Formula/px.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class Px < Formula
+  include Language::Python::Virtualenv
+
+  desc "Ps and top for human beings (px / ptop)"
+  homepage "https://github.com/walles/px"
+  url "https://github.com/walles/px.git",
+    using:    :git,
+    tag:      "1.4.0",
+    revision: "aea41c2a67a8b83e644e11d8ac152e2c123d0ee1"
```
Comment (carlocab) [inline Formula/px.rb:9]:
```suggestion
      tag:      "1.4.0",
      revision: "aea41c2a67a8b83e644e11d8ac152e2c123d0ee1"
```
We don't need `using: :git`. Also, let's align the arguments to `url`, please.

File: Formula/px.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  license "MIT"
+
+  depends_on "python@3.9"
```
Comment (carlocab) [inline Formula/px.rb:12]:
```suggestion
  depends_on "python@3.9"
  depends_on "six"
```

File: Formula/px.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  uses_from_macos "lsof"
+
+  # For updates: https://pypi.org/project/six/
+  resource "six" do
+    url "https://files.pythonhosted.org/packages/71/39/171f1c67cd00715f190ba0b100d606d440a28c93c7714febeca8b79af85e/six-1.16.0.tar.gz"
+    sha256 "1e61c37477a1626458e36f7b1d82aa5c9b094fa4802892072e49de9c60c4c926"
+  end
```
Comment (carlocab) [inline Formula/px.rb:20]:
```suggestion
```

File: Formula/px.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    url "https://files.pythonhosted.org/packages/be/ed/5bbc91f03fa4c839c4c7360375da77f9659af5f7086b7a7bdda65771c8e0/python-dateutil-2.8.1.tar.gz"
+    sha256 "73ebfe9dbf22e832286dafa60473e4cd239f8592f699aa5adaf10050e6e1823c"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    assert_equal version.to_s, pipe_output("#{bin}/px --version").strip
```
Comment (carlocab) [inline Formula/px.rb:33]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Comment by ritzmann** [comment]
Thank you @carlocab . This should address your points.

**Pull request closed without merging**